src/api.py

```diff
@@ -204,6 +204,10 @@ def do_POST(self):
             self.wfile.flush()
             self.connection.shutdown(1)
 
+            # actually handle shutdown command after sending response
+            if state.shutdown is False:
```

Comment:
I think this is a misunderstanding. It just signals to the peer that you're done writing to the socket. `man 2 shutdown`